test/controllers/opensearch_controller_test.exs

```diff
@@ -0,0 +1,12 @@
+defmodule HexWeb.OpensearchControllerTest do
+  use HexWeb.ConnCase, async: true
+
+  test "opensearch" do
+    conn = get build_conn(), "/hexsearch.xml"
+
+    path          = Path.join([__DIR__, "..", "fixtures"])
+    expected_body = File.read!(Path.join(path, "opensearch.xml"))
+
+    assert response(conn, 200) == expected_body
```

Comment:
We don't need to test the exact contents. You can use `=~` to match against a subset.